manager/integration/tests/test_ha.py

```diff
@@ -2616,7 +2616,88 @@ def test_autosalvage_with_data_locality_enabled():
     Cleaning up:
     1. Clean up the node tag
     """
-    pass
+
+    # Step1
+    nodes = client.list_node()
+    assert len(nodes) == 3
+    node_1, node_2, node_3 = nodes
+    tags = ["node-1"]
+    node_1 = common.set_node_tags(client, node_1, tags)
+
+    # Step2
+    client.create_volume(
+        name=volume_name, size=SIZE, numberOfReplicas=1,
+        nodeSelector=tags, dataLocality="best-effort"
+        )
+
+    volume = common.wait_for_volume_detached(client, volume_name)
+    assert volume.nodeSelector == tags
+
+    # Step3
+    pvc_name = volume_name + "-pvc"
+    create_pv_for_volume(client, core_api, volume, volume_name)
+    create_pvc_for_volume(client, core_api, volume, pvc_name)
+
+    # Step4
+    deployment_name = volume_name + "-dep"
+    deployment = make_deployment_with_pvc(deployment_name, pvc_name)
+    deployment["spec"]["template"]["spec"]["nodeSelector"] \
+        = {"kubernetes.io/hostname": node_2.name}
+
+    # Step5
+    apps_api = get_apps_api_client()
+    create_and_wait_deployment(apps_api, deployment)
+
+    pod_names = common.get_deployment_pod_names(core_api, deployment)
+    test_data = generate_random_data(VOLUME_RWTEST_SIZE)
+    write_pod_volume_data(core_api, pod_names[0], test_data, filename='test')
+    create_snapshot(client, volume_name)
+
+    # Step6
+    node_1_name = node_1["name"]
+    ret = core_api.list_namespaced_pod(namespace="longhorn-system")
+    for item in ret.items:
+        if "instance-manager-r" in item.metadata.name:
+            if item.metadata.labels["longhorn.io/node"] == node_1_name:
```

Comment:
You can try to use `label_selector` in `list_namespaced_pod`, so you don't need to have this forloop to find out the matched IM pod.

Reply:
HI @innobead ,
I modified code to use label_selector to find specific pod without iterate by for-loop